lib/helpers.js

```diff
@@ -209,8 +209,28 @@ function buildStartCmd (startAppOptions, apiLevel) {
   return cmd;
 }
 
+async function getSdkToolsVersion () {
+  const androidHome = process.env.ANDROID_HOME;
+  if (!androidHome) {
+    log.errorAndThrow('ANDROID_HOME environment varibale is expected to be set');
+  }
+  const propertiesPath = path.resolve(androidHome, 'tools', 'source.properties');
+  if (!await fs.exists(propertiesPath)) {
+    log.warn(`Cannot find ${propertiesPath} file to read SDK version from`);
+    return;
+  }
+  const propertiesContent = await fs.readFile(propertiesPath, 'utf8');
+  const versionMatcher = new RegExp(/Pkg\.Revision=(\d+)\.?(\d+)?\.?(\d+)?/);
+  const match = versionMatcher.exec(propertiesContent);
+  if (match) return {
```

Comment:
good point, it is a form of short-circuiting. but the exception to the rule was designed specifically for this boilerplate:

```js
if (err) {
  return cb(err);
}
```

which is super common when using callbacks. now we don't use callbacks anymore so no real need for an exception.